test/integration/migrate/index.js

```diff
@@ -34,6 +44,28 @@ module.exports = function(knex) {
         return knex.migrate.latest({directory: 'test/integration/migrate/test'}).catch(function() {});
       });
 
+      it('should remove the record in the lock table once finished', function() {
+        return knex('knex_migrations_lock').select('*').then(function(data) {
+          expect(data).to.have.length(0);
+        });
+      });
+
+      it('should throw error if the migrations are already running', function() {
+        return knex('knex_migrations_lock')
+          .insert({ is_locked: true })
+          .then(function() {
+            return knex.migrate.latest({directory: 'test/integration/migrate/test'})
+              .then(function() {
+                throw new Error('then should not execute');
+              });
+          })
+          .catch(function(error) {
+            expect(error).to.have.property('message', 'migrations failed: migration in progress');
+            // Clean up lock for other tests
+            return knex('knex_migrations_lock').del();
+          });
+      });
+
```

Comment:
Maybe one more test showing that if migration fails, but not because of locking it should still release the lock.